Can't implement traits in declarative macros.

Implementing traits doesn't seem to work either in declarative macros. [This code](https://play.rust-lang.org/?gist=85411af7902fe1202901b7448c1cba7d&version=nightly) looks like it should compile:

```rust
#![feature(decl_macro)]

pub macro create_struct($a:ident) {
    struct $a;
    impl Clone for $a {
        fn clone(&self) -> Self {
            $a
        }
    }
}

fn main() {
    create_struct!(Test);
    Test.clone();
}
```

But it does this instead:

```
   Compiling playground v0.0.1 (file:///playground)
error[E0046]: not all trait items implemented, missing: `clone`
  --> src/main.rs:5:5
   |
5  | /     impl Clone for $a {
6  | |         fn clone(&self) -> Self {
7  | |             $a
8  | |         }
9  | |     }
   | |_____^ missing `clone` in implementation
...
13 |       create_struct!(Test);
   |       --------------------- in this macro invocation
   |
   = note: `clone` from trait: `fn(&Self) -> Self`

error: aborting due to previous error

error: Could not compile `playground`.

To learn more, run the command again with --verbose.
```

--- a/src/test/run-pass/hygiene/issue-44128.rs
+++ b/src/test/run-pass/hygiene/issue-44128.rs
@@ -1,0 +1,25 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(decl_macro)]
+
+pub macro create_struct($a:ident) {
+    struct $a;
+    impl Clone for $a {
+        fn clone(&self) -> Self {
+            $a
+        }
+    }
+}
+
+fn main() {
+    create_struct!(Test);
+    Test.clone();
+}

